openlibrary/plugins/upstream/checkins.py

```diff
@@ -223,6 +223,10 @@ def POST(self):
             if goal == 0:
                 # Delete goal if "0" was submitted:
                 YearlyReadingGoals.delete_by_username_and_year(username, i.year)
+            elif goal > 2147483647:
+                raise web.badrequest(
+                    message="Reading goal cannot exceed 2147483647 books."
+                )
```

Comment:
All other validations are happening earlier in this `POST` function.  This can be checked right after `goal` is defined: https://github.com/internetarchive/openlibrary/blob/112cf1468a0ead2f2e90294596c7608dcb58c43f/openlibrary/plugins/upstream/checkins.py#L202